internal/data/common.go

```diff
@@ -232,11 +227,25 @@ type AttributeMap struct {
 	orig *[]*otlpcommon.AttributeKeyValue
 }
 
-// NewAttributeMap creates a new AttributeMap from the given map[string]AttributeValue.
-func NewAttributeMap(attrMap map[string]AttributeValue) AttributeMap {
+// NewAttributeMap creates a AttributeMap with 0 elements.
+func NewAttributeMap() AttributeMap {
+	orig := []*otlpcommon.AttributeKeyValue(nil)
+	return AttributeMap{&orig}
+}
+
+func newAttributeMap(orig *[]*otlpcommon.AttributeKeyValue) AttributeMap {
+	return AttributeMap{orig}
+}
+
+// InitFromMap overwrites the entire AttributeMap and reconstructs the AttributeMap
+// with values from the given map[string]string.
+//
+// Returns the same instance to allow nicer code like:
+// assert.EqualValues(t, NewAttributeMap().InitFromMap(map[string]AttributeValue{...}), actual)
+func (am AttributeMap) InitFromMap(attrMap map[string]AttributeValue) AttributeMap {
```

Comment:
Nit: Should we `CopyFromX` naming for these operations?

Reply:
As I mentioned in the description we will redesign a bit the API to create a new map (especially from OC case). For the moment I will keep it this way until we do that test.